insonmnia/node/server.go

```diff
@@ -94,42 +94,40 @@ func New(ctx context.Context, c Config, key *ecdsa.PrivateKey) (*Node, error) {
 	}
 
 	creds := util.NewTLS(TLSConfig)
-	srv := util.MakeGrpcServer(creds)
 
 	opts, err := newRemoteOptions(ctx, key, c, creds)
 	if err != nil {
 		return nil, err
 	}
 
-	// register hub connection if hub addr is set
-	if c.HubEndpoint() != "" {
-		hub, err := newHubAPI(opts)
-		if err != nil {
-			return nil, err
-		}
-		pb.RegisterHubManagementServer(srv, hub)
-		log.G(ctx).Info("hub service registered", zap.String("endpt", c.HubEndpoint()))
-	}
+	hub := newHubAPI(opts)
 
 	market, err := newMarketAPI(opts)
 	if err != nil {
 		return nil, err
 	}
-	pb.RegisterMarketServer(srv, market)
-	log.G(ctx).Info("market service registered", zap.String("endpt", c.MarketEndpoint()))
 
 	deals, err := newDealsAPI(opts)
 	if err != nil {
 		return nil, err
 	}
-	pb.RegisterDealManagementServer(srv, deals)
-	log.G(ctx).Info("deals service registered")
 
 	tasks, err := newTasksAPI(opts)
 	if err != nil {
 		return nil, err
 	}
 
+	srv := util.MakeGrpcServer(creds, grpc.UnaryInterceptor(hub.(*hubAPI).intercept))
```

Comment:
Looks rather strange that you use hubAPI interceptor for all services

Reply:
According to [docs](https://godoc.org/google.golang.org/grpc#UnaryInterceptor), it unable to register more that one UnaryInterceptor for the gPRC server. Or may I do it wrong and the handler needs to be registered in another way?